packages/kbn-content-management-table-list/src/table_list_view/services.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FC, useContext } from 'react';
+import type { Observable } from 'rxjs';
+import type { EuiTableFieldDataColumnType, SearchFilterConfig } from '@elastic/eui';
+import type { CoreTheme } from '@kbn/core-theme-browser';
+
+import type { SavedObject, MountPoint, SavedObjectsFindOptionsReference } from '../types';
+
+/**
+ * Abstract external services for this component.
+ */
+export interface Services {
+  application: {
```

Comment:
We should talk about this in person, but basically you're running into trouble here because you're allowing the shape of Kibana dependencies to influence the shape of your components' services.  Consider `NoDataViewsPrompt`:

https://github.com/elastic/kibana/blob/main/packages/shared-ux/prompt/no_data_views/src/services.tsx

By designing an "services" API, we can identify what functionality the components are using into discrete functions.  Then, we can implement those functions with Kibana services, and those impls do not have to be 1:1-- we can utilize a lot of Kibana services in the component services API and keep that logic out of the component.

Let's get some time on the calendar soon... our conversation can help me with the documentation I'm writing on precisely this subject.

Reply:
Yes let's discuss it in person to be sure I understand what you mean. I thought that we could help the consumers and let them provide the plugin start contract "as is" without asking each and everyone to write a function to satisfy the package service. Or I am misunderstanding what you are saying.

As you can see, I've only asked in the services the handlers that the component package actually requires.